include/NovelRT/Utilities/Event.h

```diff
@@ -53,25 +58,37 @@ namespace NovelRT::Utilities {
       return _handlers.size();
     }
 
-    void operator+=(const EventHandler<TArgs...>& handler) {
+    Atom operator+=(const EventHandler<TArgs...>& handler) {
       if (handler.getId() != Atom()) {
         _handlers.emplace_back(handler);
       }
+      return handler.getId();
+    }
+
+    Atom operator+=(const std::function<void(TArgs...)>& function) {
+      auto handler = EventHandler<TArgs...>(function);
+      return *this += handler;
     }
 
-    void operator+=(const std::function<void(TArgs...)>& function) {
+    Atom operator+=(std::function<void(TArgs...)>&& function) {
       auto handler = EventHandler<TArgs...>(function);
-      *this += handler;
+      return *this += handler;
     }
 
     void operator-=(const EventHandler<TArgs...>& handler) {
-      if (handler.getId() == Atom())
-        return;
+      remove(handler.getId());
+    }
 
-      auto match = std::find(_handlers.cbegin(), _handlers.cend(), handler);
+    void remove(Atom eventHandlerId) {
+      if (eventHandlerId == Atom())
+        return;
```

Comment:
I'm not entirely sure I understand this?

Reply:
this was an adaptation of the same code before, so event handlers that have the default Atom (because they are empty) don't do anything